core/src/main/java/org/infinispan/protostream/impl/TagWriterImpl.java

```diff
@@ -915,4 +931,295 @@ void flush() throws IOException {
+         int tagLen = varIntBytes(tag);
+         int valueLen = varIntBytes(value);
+         out.ensureCapacity(pos + tagLen + valueLen);
+
+         byte[] buf = out.getRawBuffer();
+         pos = writeVarInt32Direct(buf, pos, tag);
+         pos = writeVarInt64Direct(buf, pos, value);
+         out.setPosition(pos);
+      }
+
+      @Override
+      void writeFixed32Field(int fieldNumber, int value) {
+         int pos = out.getPosition();
+         int tag = WireType.makeTag(fieldNumber, WireType.WIRETYPE_FIXED32);
+         int tagLen = varIntBytes(tag);
+         out.ensureCapacity(pos + tagLen + FIXED_32_SIZE);
+
+         byte[] buf = out.getRawBuffer();
+         pos = writeVarInt32Direct(buf, pos, tag);
+         pos = writeFixed32Direct(buf, pos, value);
+         out.setPosition(pos);      }
+
+      @Override
+      void writeFixed64Field(int fieldNumber, long value) {
+         int pos = out.getPosition();
+         int tag = WireType.makeTag(fieldNumber, WireType.WIRETYPE_FIXED64);
+         int tagLen = varIntBytes(tag);
+         out.ensureCapacity(pos + tagLen + FIXED_64_SIZE);
+
+         byte[] buf = out.getRawBuffer();
+         pos = writeVarInt32Direct(buf, pos, tag);
+         pos = writeFixed64Direct(buf, pos, value);
+         out.setPosition(pos);
+      }
+
+      @Override
+      void writeBoolField(int fieldNumber, boolean value) {
+         int pos = out.getPosition();
+         int tag = WireType.makeTag(fieldNumber, WireType.WIRETYPE_VARINT);
+         int tagLen = varIntBytes(tag);
+         out.ensureCapacity(pos + tagLen + 1);
+
+         byte[] buf = out.getRawBuffer();
+         pos = writeVarInt32Direct(buf, pos, tag);
+         buf[pos++] = (byte) (value ? 1 : 0);
+         out.setPosition(pos);
+      }
+
+      @Override
+      void writeLengthDelimitedField(int fieldNumber, int length) {
+         int pos = out.getPosition();
+         int tag = WireType.makeTag(fieldNumber, WireType.WIRETYPE_LENGTH_DELIMITED);
+         int tagLen = varIntBytes(tag);
+         int valueLen = varIntBytes(length);
+         out.ensureCapacity(pos + tagLen + valueLen);
+
+         byte[] buf = out.getRawBuffer();
+         pos = writeVarInt32Direct(buf, pos, tag);
+         pos = writeVarInt32Direct(buf, pos, length);
+         out.setPosition(pos);
+      }
+
+      @Override
+      void writeBytes(int fieldNumber, ByteBuffer value) {
+         int pos = out.getPosition();
+         if (value.hasArray()) {
+            int length = value.remaining();
+            writeBytes(fieldNumber, value.array(), value.arrayOffset() + value.position(), length);
+            value.position(value.position() + length);
+         } else {
+            int bbPos = value.position();
+            int bbLen = bbPos + value.remaining();
+
+            int tag = WireType.makeTag(fieldNumber, WireType.WIRETYPE_LENGTH_DELIMITED);
+            out.ensureCapacity(pos + MAX_VARINT_SIZE + MAX_VARINT_SIZE + bbLen);
```

Comment:
I'm wondering why in some places you use `varIntBytes` and in another `MAX_VARINT_SIZE`.

Reply:
Something I missed with all the refactoring :upside_down_face: